SimpleEdgeDiscovery r2.2 Fall25 M4

Kevsy opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* subproject management



#### What this PR does / why we need it:

Publication of Fall'25 M4 release r2.2 of SimpleEdgeDiscovery v2.0.0


#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes # N/A 

#### Special notes for reviewers:



#### Changelog input

```
 release-note
 - Publication of Fall'25 M4 public release of simple-edge-discovery v0.2.0

```

#### Additional documentation 

This section can be blank.



```
docs

```

Kevsy committed:
chore: update CHANGELOG.md with M4 r2.2 values

Kevsy committed:
chore: update release values in API YAML

Also corrected two minor typos (no semantic change)

Kevsy committed:
chore: update release values in links and Feature header

Kevsy committed:
chore: updated release values and added Test Statement issue link

Kevsy committed:
chore: update release numbers and NEW public release statement

maheshc01 commented on CHANGELOG.md:
```diff
@@ -23,14 +24,14 @@ The below sections record the changes for each API version in each release as fo
 
 NOTE: SimpleEdgeDiscovery is part of the EdgeCloud API Sub Project, and was previously developed within the [EdgeCloud](https://github.com/camaraproject/EdgeCloud) repository. Two releases were made within EdgeCloud, v0.81 and r0.9.3-rc. These releases also include the other EdgeCloud APIs and materials, so the changelog below highlights only the changes related to SimpleEdgeDiscovery. Further in ...
 
-# r2.1 - rc
```
I see that r2.1 has been replaced with r2.2. 
my understanding is that r2.1 needs to be retained and a new section for r2.2 needs to be added. like how the table of contents has it.

Kevsy replied:
Thanks Mahesh - good catch, I've put it back now.

Kevsy committed:
fix: re-included the r2.1 pre-release details

maheshc01 commented:
looks good

Kevsy commented:
Note to reviewers: I will rebase this M4 PR to account for the fix to #130 (ErrorInfo order reversed) - please wait to review until I have confirmed that has been completed.

Kevsy committed:
Merge branch 'camaraproject:main' into Kevsy-r2.2

Kevsy commented:
> Note to reviewers: I will rebase this M4 PR to account for the fix to #130 (ErrorInfo order reversed) - please wait to review until I have confirmed that has been completed.

OK that's now fixed, but @rartych is going to update the linting first before further review by release-management-reviewers.

Kevsy commented:
OK all good now for the release-management-reviewer to proceed 👍

tanjadegroot commented on CHANGELOG.md:
```diff
@@ -4,6 +4,7 @@ NOTE:
 
 ## Table of contents
 
+- [r2.2](#r22)
```
```suggestion
- [r2.2](#r22---(Fall25-public-release))
```

tanjadegroot replied:
you need to put the full title in the index else the link will not work.

tanjadegroot commented on code/Test_definitions/simple-edge-discovery.feature:
no test scenarios for the errors 400 and 401 in the .feature fil
the last scenario on test for error 500 internal can be removed

tanjadegroot reviewed (changes requested):
Some error cases are not covered by any tests, for a v2 API I think these should be added, but up to team to decide if to do now or for the next release.

One small fix needed in the Changelog. 
all the rest is ok

Kevsy committed:
fix: add r2.2 title to link in CHANGELOG.md

Co-authored-by: Tanja de Groot <87864067+tanjadegroot@users.noreply.github.com>

Kevsy commented:
Thanks @tanjadegroot - CHANGELOG.MD fix made.

> Some error cases are not covered by any tests, for a v2 API I think these should be added, but up to team to decide if to do now or for the next release.
> 
> One small fix needed in the Changelog. all the rest is ok

I have created PR #135 to add the 400 and 401 scenarios/remove the #500/tidy up. If a codeowner can approve in time then I can merge into the M4 PR.

Kevsy committed:
Merge branch 'camaraproject:main' into Kevsy-r2.2

Kevsy commented:
HI @tanjadegroot , I've updated the errors in a separate PR which has now been merged to wip main and then synched into the M4 PR. So if you expand the .`feature` file in the M4 PR code you will see new 400 and 401 errors and the 500 removed.

tanjadegroot commented on CHANGELOG.md:
```diff
@@ -4,6 +4,7 @@ NOTE:
 
 ## Table of contents
 
+- [r2.2](#r22---(Fall25-public-release))
```
@Kevsy  the link does not work:

option 1) try
line 7: [r2.2](#r22-(Fall25-public-release))
or
line 7: [r2.2](#r22-\(Fall25-public-release\))

option 2) if above does not work: remove the parentheses from line 27 and add a "-", and update the index as follows:

line 7: [r2.2](#r22---Fall25-public-release) 
line 27: r2.2 - Fall25 public release

Kevsy replied:
```suggestion
- [r2.2](#r22---Fall25-public-release)
```

tanjadegroot reviewed (changes requested):
remains an issue with the index in the CHANGELOG file
all the rest is OK

Please try the fixes I propose and if one of them work (option 2 should work), all is OK.

tanjadegroot commented:
> HI @tanjadegroot , I've updated the errors in a separate PR which has now been merged to wip main and then synched into the M4 PR. So if you expand the .`feature` file in the M4 PR code you will see new 400 and 401 errors and the 500 removed.

@Kevsy : I do not see the updates in the M4 release PR ?

BTW: there seems to be an error I saw in the merged PR (#135) in the .feature file:
- line 33 references 404
- line 37 references 422
- line 40 references 400 (btw this is also in other 422 scenarios)

Kevsy commented on CHANGELOG.md:
```diff
@@ -23,6 +24,48 @@ The below sections record the changes for each API version in each release as fo
 
 NOTE: SimpleEdgeDiscovery is part of the EdgeCloud API Sub Project, and was previously developed within the [EdgeCloud](https://github.com/camaraproject/EdgeCloud) repository. Two releases were made within EdgeCloud, v0.81 and r0.9.3-rc. These releases also include the other EdgeCloud APIs and materials, so the changelog below highlights only the changes related to SimpleEdgeDiscovery. Further in ...
 
+# r2.2 (Fall25 public release)
```
```suggestion
# r2.2 - Fall25 public release
```

Kevsy committed:
Update CHANGELOG.md

Kevsy committed:
Update CHANGELOG.md

Kevsy committed:
fix: corrected error codes in .feature

Kevsy committed:
fix: IDENTIFIER_NOT_FOUND change changed to 404 code

Kevsy commented:
Thanks @tanjadegroot  - I can see the .feature changes in the M4 PR...now updated to include your suggestions (thx!), note I have corrected the line 37 block to be a 404 

CHANGELOG updated too per option 2.

tanjadegroot reviewed (approved):
@Kevsy thanks for the updates - all good now.

LGTM from Release Management

Kevsy commented:
Thanks @tanjadegroot ! @maheshc01 please can you re-approve?

maheshc01 commented:
Approved!
